packages/core-mobile/app/new/routes/onboarding/mnemonic/createPin.tsx

```diff
@@ -33,6 +36,21 @@ export default function CreatePin(): JSX.Element {
     [mnemonic, navigate, onPinCreated, useBiometrics]
   )
 
+  useEffect(() => {
+    BiometricsSDK.canUseBiometry()
+      .then((biometricAvailable: boolean) => {
+        setIsBiometricAvailable(biometricAvailable)
+      })
+      .catch(Logger.error)
+
+    const type = commonStorage.getString(StorageKey.SECURE_ACCESS_SET)
+    if (type) {
+      setUseBiometrics(type === 'BIO')
```

Comment:
looks like we are doing this same logic in 3-4 different places. can this logic be moved DeviceInfoService so we can cache the result? also if you can create a hook for this as well, that will be nice.